packages/kcms/src/match/service/generatePre.test.ts

```diff
@@ -1,71 +1,157 @@
 import { config } from 'config';
 import { describe, expect, it } from 'vitest';
 import { SnowflakeIDGenerator } from '../../id/main';
 import { DummyRepository } from '../../team/adaptor/repository/dummyRepository';
-import { TeamID } from '../../team/models/team';
+import { Team, TeamID } from '../../team/models/team';
 import { FetchTeamService } from '../../team/service/fetchTeam';
 import { testTeamData } from '../../testData/entry';
 import { DummyPreMatchRepository } from '../adaptor/dummy/preMatchRepository';
 import { GeneratePreMatchService } from './generatePre';
 
-describe('GeneratePreMatchService', () => {
-  const teamRepository = new DummyRepository([...testTeamData.values()]);
+const createGenerateService = (teamData: Team[]) => {
+  const teamRepository = new DummyRepository(teamData);
   const fetchService = new FetchTeamService(teamRepository);
   const generator = new SnowflakeIDGenerator(1, () =>
     BigInt(new Date('2024/01/01 00:00:00 UTC').getTime())
   );
   const preMatchRepository = new DummyPreMatchRepository();
   const generateService = new GeneratePreMatchService(fetchService, generator, preMatchRepository);
 
-  const expectedTeamPair = [
-    [
-      ['A1', 'B3'],
-      ['A4', 'N1'],
-      ['B3', 'A1'],
-      ['N1', 'A4'],
-    ],
-    [
-      ['A2', 'C1'],
-      ['B1', 'N2'],
-      ['C1', 'A2'],
-      ['N2', 'B1'],
-    ],
-    [
-      ['A3', 'C2'],
-      ['B2', undefined],
-      [undefined, 'A3'],
-      ['C2', 'B2'],
-    ],
-  ];
-
-  it('正しく予選対戦表を生成できる', async () => {
-    const generated = await generateService.handle('elementary');
+  return { generateService, preMatchRepository };
+};
+
+describe('GeneratePreMatchService', () => {
+  it('正しく予選対戦表を生成できる - 部門ごと', async () => {
+    const { generateService, preMatchRepository } = createGenerateService([
+      ...testTeamData.values(),
+    ]);
+
+    const generated = await generateService.generateByDepartment('elementary');
     expect(Result.isOk(generated)).toBe(true);
     const res = Result.unwrap(generated);
 
-    for (let i = 0; i < config.match.pre.course.elementary.length; i++) {
-      const course = res.filter((v) => v.getCourseIndex() === i + 1);
-      const pair = course.map((v) => [
-        testTeamData.get(v.getTeamID1() ?? ('' as TeamID))?.getTeamName(),
-        testTeamData.get(v.getTeamID2() ?? ('' as TeamID))?.getTeamName(),
-      ]);
-      expect(pair).toStrictEqual(expectedTeamPair[i]);
+    const leftTeamNames: (string | undefined)[] = [];
+    const rightTeamNames: (string | undefined)[] = [];
+
+    for (const match of res) {
+      // ペアが同じチーム同士でない
+      const leftTeamName = testTeamData.get(match.getTeamID1() ?? ('' as TeamID))?.getTeamName();
+      const rightTeamName = testTeamData.get(match.getTeamID2() ?? ('' as TeamID))?.getTeamName();
+      expect(leftTeamName).not.toStrictEqual(rightTeamName);
+
+      if (leftTeamName) {
+        leftTeamNames.push(leftTeamName);
+      }
```

Comment:
JSでは `""` ( 空文字列 ) も Falsy なので、 `undefined` だけをはじくには
```suggestion
      if (leftTeamName != null) leftTeamNames.push(leftTeamName);
```
とするべきです。